src/vsr/grid.zig

```diff
@@ -330,20 +380,20 @@ pub fn GridType(comptime Storage: type) type {
             // Now that there are no writes to released blocks, we can safely mark them as free.
             // This concludes grid checkpointing.
             grid.free_set.checkpoint(
-                grid.free_set_encoded.block_addresses[0..grid.free_set_encoded.block_count()],
+                grid.free_set_checkpoint.block_addresses[0..grid.free_set_checkpoint.block_count()],
             );
-            assert(grid.free_set.count_released() == grid.free_set_encoded.block_count());
+            assert(grid.free_set.count_released() == grid.free_set_checkpoint.block_count());
 
-            const callback = grid.checkpointing.?.callback;
-            grid.checkpointing = null;
+            const callback = grid.callback.checkpoint;
+            grid.callback = .none;
             callback(grid);
         }
 
         pub fn cancel(grid: *Grid, callback: *const fn (*Grid) void) void {
-            assert(grid.canceling == null);
-            assert(grid.checkpointing == null);
+            // grid.open() is cancellable the same way that read_block()/write_block() are.
+            assert(grid.callback == .none or grid.callback == .open);
```

Comment:
maybe switch for exhaustiveness here? feels useful to decect missing _new_ cancelation cases at compiletime.